frontend/src/components/modals/NotifyUsersModal.tsx

```diff
@@ -0,0 +1,223 @@
+import { RoleType } from '../../../../shared/types/customTypes';
+import css from './NotifyUsersModal.module.scss';
+import { getType } from '../../utils/UserUtils';
+import { Modal, ModalTypes } from './types';
+import {
+  findMissingCustomers,
+  findMissingDevelopers,
+} from '../../utils/TaskUtils';
+
+const classes = classNames.bind(css);
+
+interface CheckableUser extends RoadmapUser {
+  checked: boolean;
+}
+
+export const NotifyUsersModal: Modal<ModalTypes.NOTIFY_USERS_MODAL> = ({
+  closeModal,
+  taskId,
+}) => {
+  const dispatch = useDispatch<StoreDispatchType>();
+  const task = useSelector(taskSelector(taskId))!;
+  const [isLoading, setIsLoading] = useState(false);
+  const userInfo = useSelector<RootState, UserInfo | undefined>(
+    userInfoSelector,
+    shallowEqual,
+  );
+  const customers = useSelector<RootState, Customer[] | undefined>(
+    allCustomersSelector(),
+    shallowEqual,
+  );
+  const allUsers = useSelector<RootState, RoadmapUser[] | undefined>(
+    roadmapUsersSelector(),
+    shallowEqual,
+  );
+  const [missingUsers, setMissingUsers] = useState<CheckableUser[] | undefined>(
+    [],
+  );
+  const [previousUsers, setPreviousUsers] = useState<
+    CheckableUser[] | undefined
+  >([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [allChecked, setAllChecked] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (
+      getType(userInfo?.roles, task.roadmapId) === RoleType.Admin &&
+      customers &&
+      allUsers
+    ) {
+      const missingCustomers = findMissingCustomers(task.ratings, customers);
+      const missingRepresentatives = new Map<number, CheckableUser>();
+
+      // Find representatives who haven't given ratings
+      if (missingCustomers) {
+        missingCustomers.forEach((customer) => {
+          // Filter out logged in user, so they won't send email to themselves
+          const representatives = customer?.representatives?.filter(
+            (user) => user.id !== userInfo?.id,
+          );
+          const ratingsForTask = task.ratings
+            .filter((rating) => rating.forCustomer === customer.id)
+            .map((e) => e.createdByUser);
+
+          if (representatives) {
+            representatives.forEach((rep) => {
+              if (!ratingsForTask.includes(rep.id))
+                if (
+                  !Array.from(missingRepresentatives.values()).some(
+                    (e) => e.id === rep.id,
+                  )
+                )
+                  missingRepresentatives.set(rep.id, {
+                    ...rep,
+                    checked: false,
+                  });
+            });
+          }
+        });
+      }
+
+      const missingDevelopers = findMissingDevelopers(
+        task.ratings,
+        allUsers,
+      ).map((dev) => ({ ...dev, checked: false }));
+
+      setMissingUsers([
+        ...(Array.from(missingRepresentatives.values()) ?? []),
+        ...(missingDevelopers ?? []),
```

Comment:
Neither will be undefined
```suggestion
        ...Array.from(missingRepresentatives.values()),
        ...missingDevelopers,
```